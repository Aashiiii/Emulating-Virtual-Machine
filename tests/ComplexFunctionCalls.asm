FUNC LABEL 0
    MOV REG 0 VAL 3
    MOV STK A REG 0
    MOV REG 1 VAL 4
    ADD REG 0 REG 1
    CAL VAL 2
    CAL VAL 4
    CAL VAL 3
    RET
FUNC LABEL 2
    MOV REG 0 VAL 7
    ADD REG 0 REG 1
    CAL VAL 7
    PRINT REG 0
    CAL VAL 5
    RET
FUNC LABEL 4
    MOV REG 1 VAL 3
    MOV STK A REG 1
    PRINT STK A
    NOT REG 1
    PRINT REG 1
    CAL VAL 7
    PRINT VAL 0
    CAL VAL 5
    RET
FUNC LABEL 3
    ADD REG 0 REG 1
    PRINT REG 0
    PRINT REG 1
    CAL VAL 7
    CAL VAL 7
    CAL VAL 5
    RET
FUNC LABEL 7
    MOV STK A REG 0
    PRINT STK A
    ADD REG 0 REG 1
    PRINT REG 0
    RET
FUNC LABEL 5
    PRINT REG 0
    PRINT REG 1
    RET
