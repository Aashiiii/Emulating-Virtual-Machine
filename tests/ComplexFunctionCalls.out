11
15
15
15
4
3
252
15
11
0
11
252
7
252
7
3
3
255
255
252
